src/contracts/LongExecutor.sol

```diff
@@ -0,0 +1,1325 @@
+        VOTE_DIFFERENTIAL
+      ));
+  }
+}
+
+
+interface IExecutor {
+  /**
+  * @dev method tu update the voting duration of the proposal. Only callable by admin.
+  * @param votingDuration duration of the vote
+  */
+  function updateVotingDuration(uint256 votingDuration) external;
+
+  /**
+  * @dev method to update the vote differential needed to pass the proposal. Only callable by admin.
+  * @param voteDifferential differential needed on the votes to pass the proposal
+  */
+  function updateVoteDifferential(uint256 voteDifferential) external;
+
+  /**
+  * @dev method to update the minimum quorum needed to pass the proposal. Only callable by admin.
+  * @param minimumQuorum quorum needed to pass the proposal 
+  */
+  function updateMinimumQuorum(uint256 minimumQuorum) external;
+  /**
+    * @dev method to update the propositionThreshold. Only callable by admin.
+    * @param propositionThreshold new proposition threshold
+    **/
+  function updatePropositionThreshold(uint256 propositionThreshold) external;
+
+  // event triggered when voting duration gets updated by the admin
+  event VotingDurationUpdated(address indexed executor, address indexed admin, uint256 oldVotingDuration, uint256 newVotingDuration);
+  // event triggered when vote differential gets updated by the admin
+  event VoteDifferentialUpdated(address indexed executor, address indexed admin, uint256 oldVoteDifferential, uint256 newVoteDifferential);
+  // event triggered when minimum quorum gets updated by the admin
+  event MinimumQuorumUpdated(address indexed executor, address indexed admin, uint256 oldMinimumQuorum, uint256 newMinimumQuorum);
+  // event triggered when proposition threshold gets updated by the admin
+  event PropositionThresholdUpdated(address indexed executor, address indexed admin, uint256 oldPropositionThreshold, uint256 newPropositionThreshold);
+}
+
+/**
+ * @title Time Locked, Validator, Executor Contract
+ * @dev Contract
+ * - Validate Proposal creations/ cancellation
+ * - Validate Vote Quorum and Vote success on proposal
+ * - Queue, Execute, Cancel, successful proposals' transactions.
+ * @author Aave
+ **/
+contract Executor is ExecutorWithTimelock, ProposalValidator, IExecutor {
+  constructor(
+    address admin,
+    uint256 delay,
+    uint256 gracePeriod,
+    uint256 minimumDelay,
+    uint256 maximumDelay,
+    uint256 propositionThreshold,
+    uint256 voteDuration,
+    uint256 voteDifferential,
+    uint256 minimumQuorum
+  )
+    ExecutorWithTimelock(admin, delay, gracePeriod, minimumDelay, maximumDelay)
+    ProposalValidator(propositionThreshold, voteDuration, voteDifferential, minimumQuorum)
+  {}
+
+  /// @inheritdoc IExecutor
+  function updateVotingDuration(uint256 votingDuration) external override onlyAdmin {
+    uint256 oldVotingDuration = VOTING_DURATION;
+    VOTING_DURATION = votingDuration;
+    emit VotingDurationUpdated(address(this), getAdmin(), oldVotingDuration, VOTING_DURATION);
```

Comment:
Don't think it is necessary to emit old value, it is common practice to emit the new one.
Also, it could be important to emit the event on the first set (constructor on father). Which maybe leads to have an internal function to do everything